r2/r2/lib/providers/search/solrsearch.py

```diff
@@ -0,0 +1,742 @@
+DOC_API = 'http://%s:%s' % (g.solr_doc_host, g.solr_port)
+
+SORTS_DICT = {'-text_relevance': 'score desc',
+              'relevance': 'score desc'}
+
+def basic_query(query=None, bq=None, faceting=None, size=1000,
+                start=0, rank="", return_fields=None, record_stats=False,
+                search_api=None):
+    if search_api is None:
+        search_api = g.solr_search_host
+    if faceting is None:
+        faceting = DEFAULT_FACETS
+    path = _encode_query(query, faceting, size, start, rank, return_fields)
+    timer = None
+    if record_stats:
+        timer = g.stats.get_timer("cloudsearch_timer")
+        timer.start()
+    connection = httplib.HTTPConnection(search_api, g.solr_port)
+    try:
+        connection.request('GET', path)
+        resp = connection.getresponse()
+        response = resp.read()
+        if record_stats:
+            g.stats.action_count("event.search_query", resp.status)
+        if resp.status >= 300:
+            try:
+                reasons = json.loads(response)
+            except ValueError:
+                pass
+            else:
+                messages = reasons.get("messages", [])
+                for message in messages:
+                    if message['code'] in INVALID_QUERY_CODES:
+                        raise InvalidQuery(resp.status, resp.reason, message,
+                                           path, reasons)
+            raise SearchHTTPError(resp.status, resp.reason, path,
+                                       response)
+    finally:
+        connection.close()
+        if timer is not None:
+            timer.stop()
+
+    return json.loads(response)
+
+
+basic_link = functools.partial(basic_query, size=10, start=0,
+                               rank="",
+                               return_fields=['title', 'reddit',
+                                              'author_fullname'],
+                               record_stats=False,
+                               search_api=g.solr_search_host)
+
+
+basic_subreddit = functools.partial(basic_query,
+                                    faceting=None,
+                                    size=10, start=0,
+                                    rank="activity",
+                                    return_fields=['title', 'reddit',
+                                                   'author_fullname'],
+                                    record_stats=False,
+                                    search_api=g.solr_subreddit_search_host)
+
+
+class SolrSearchQuery(object):
+    '''Represents a search query sent to solr'''
+    search_api = None
+    sorts = {}
+    sorts_menu_mapping = {}
+    recents = {None: None}
+    known_syntaxes = ("solr", "plain")
+    default_syntax = "solr"
+
+    def __init__(self, query, sr=None, sort=None, syntax=None, raw_sort=None,
+                 faceting=None, recent=None):
+        if syntax is None:
+            syntax = self.default_syntax
+        elif(syntax == 'cloudsearch'):
```

Comment:
I don't believe that "cloudsearch" syntax is compatible with Solr; unless that's true, this block should be removed.